xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java

```diff
@@ -42,35 +56,106 @@
-  @JsonProperty("ledger_index_max")
+  @JsonIgnore
+  @Deprecated
+  @Value.Auxiliary
   Optional<LedgerIndex> ledgerIndexMax();
 
+  /**
+   * The earliest ledger to include transactions from. A value of {@code -1} instructs the server to use the
+   * earliest validated ledger version available.
+   *
+   * @return A {@link LedgerIndexBound} with a default of empty.
+   */
+  @JsonProperty("ledger_index_min")
+  @Value.Default
+  @Nullable // Value.Default on Optional attributes takes away the non-optional builder method
+  default LedgerIndexBound ledgerIndexMinimum() {
+    // Gives deprecated field precedence
+    return ledgerIndexMin()
+      .map(LedgerIndex::unsignedIntegerValue)
+      .map(UnsignedInteger::intValue)
+      .map(LedgerIndexBound::of)
+      .orElse(LedgerIndexBound.of(-1));
+  }
+
+  /**
+   * The most recent ledger to include transactions from. A value of {@code -1} instructs the server to use the most
+   * recent validated ledger version available.
+   *
+   * @return A {@link LedgerIndexBound} with a default of empty.
+   */
+  @JsonProperty("ledger_index_max")
+  @Value.Default
+  @Nullable // Value.Default on Optional attributes takes away the non-optional builder method
+  default LedgerIndexBound ledgerIndexMaximum() {
+    // Gives deprecated field precedence
+    return ledgerIndexMax()
+      .map(LedgerIndex::unsignedIntegerValue)
+      .map(UnsignedInteger::intValue)
+      .map(LedgerIndexBound::of)
+      .orElse(LedgerIndexBound.of(-1));
+  }
+
   /**
    * Return transactions from the ledger with this hash only.
    *
    * @return An optionally-present {@link Hash256} containing the ledger hash.
+   * @deprecated Ledger hash should be specified in {@link #ledgerSpecifier()}.
    */
-  @JsonProperty("ledger_hash")
+  @JsonIgnore
+  @Deprecated
+  @Value.Auxiliary
   Optional<Hash256> ledgerHash();
 
   /**
    * Return transactions from the ledger with this index only.
    *
    * @return A {@link LedgerIndex} containing the ledger index, defaults to "current".
+   * @deprecated Ledger index and any shortcut values should be specified in {@link #ledgerSpecifier()}.
    */
-  @JsonProperty("ledger_index")
+  @JsonIgnore
+  @Deprecated
+  @Value.Auxiliary
   Optional<LedgerIndex> ledgerIndex();
 
+  /**
+   * Specifies the ledger version to request. A ledger version can be specified by ledger hash,
+   * numerical ledger index, or a shortcut value.
+   *
+   * <p>The only valid ledger index shortcut for this request object is
+   * {@link org.xrpl.xrpl4j.model.client.common.LedgerIndexShortcut#VALIDATED}.</p>
+   *
+   * <p>Setting this value will nullify and take precedence over {@link #ledgerIndexMinimum()}
+   * and {@link #ledgerIndexMaximum()}</p>
+   *
+   * @return A {@link LedgerSpecifier} specifying the ledger version to request.
+   */
+  @JsonUnwrapped
+  @Value.Default // TODO: Make non-default once ledgerIndex and ledgerHash are gone
+  default Optional<LedgerSpecifier> ledgerSpecifier() {
```

Comment:
This gives a `warn` during compile because it's unusual for an Optional method to have a default value -- can this be made non-optional?

Reply:
It has to be `Optional`, since in these request params you don't have to specify a `ledger_index` or `ledger_hash` (you can specify `ledgerIndexMinimum` or `ledgerIndexMaximum`). It's not ideal to have a `@Value.Default` for an `Optional`, but I think it's the only way to get the deprecation to work correctly. Once we get rid of the deprecated fields, we can get rid of the `Default`

Reply:
I thought from yesterday we said the default (now and in the future) would just be `current`. So, can't we just get rid of the `Optional`, but keep the `@Default`?